### ISSUE TITLE
salt-cloud is  unable to work with virtualbox

### ISSUE THREAD

**Issue by vipullinux** [saltstack/salt#51020]
### Unable to work with virtualbox
I'm trying to perform this task [Getting Started With Virtualbox](https://docs.saltstack.com/en/latest/topics/cloud/virtualbox.html)
Below is the log , after executing the command ` salt-cloud -f show_image virtualbox image=Ubuntu -l debug`
[logs.TXT](https://github.com/saltstack/salt/files/2719827/logs.TXT)

### Configuration
/etc/salt/cloud.profiles.d/virtualbox.conf
> virtualbox-test:
>     provider: virtualbox-config
>     clonefrom: Ubuntu
>     # Optional
>     power_on: True
>     deploy: True
>     ssh_username: root
>     password: 123456
>     sudo: node
>     sudo_password: 123456
>     # Example minion config
>     minion:
>         master: localhost
>     make_master: True

/etc/salt/cloud.providers.d
> virtualbox-config:
>    driver: virtualbox

### Versions Report
Salt Version:
           Salt: 2018.3.3
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.4.2
      docker-py: Not Installed
          gitdb: 0.6.4
      gitpython: 1.0.1
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.3
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.12 (default, Nov 12 2018, 14:36:49)
   python-gnupg: 0.3.8
         PyYAML: 3.11
          PyZMQ: 15.2.0
           RAET: Not Installed
          smmap: 0.9.0
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4
 
System Versions:
           dist: Ubuntu 16.04 xenial
         locale: UTF-8
        machine: x86_64
        release: 4.15.0-43-generic
         system: Linux
        version: Ubuntu 16.04 xenial

**Comment by garethgreenaway** [comment]
@vipullinux Unfortunately I haven't been able to reproduce this, mostly due to issues installing the various VirtualBox python libraries, but this is a really weird error as `reload` is a built in Python function so it should be available unless you're running Python 3.x in which case it would use `reload` from the importlib library.  @saltstack/team-core Thoughts?

**Comment by waynew** [comment]
@garethgreenaway I was able to reproduce this - I created a vagrant box from `ubuntu/xenial64`, inside the vagrant box I installed vagrant, python-minimal and virtualbox + SDK. Then I installed python-pip, then I did `sudo python2 -m pip install salt`, did

```
mkdir -p /etc/salt/cloud.providers.d /etc/salt/cloud.profiles.d
```

Then I have the following:

```
vagrant@ubuntu-xenial:/etc/salt/cloud.profiles.d$ cat /etc/salt/cloud.providers.d/virtualbox.conf
virtualbox-config:
  driver: virtualbox
vagrant@ubuntu-xenial:/etc/salt/cloud.profiles.d$ ^providers^profiles
cat /etc/salt/cloud.profiles.d/virtualbox.conf
virtualbox-test:
  provider: virtualbox-config
  clonefrom: Ubuntu
  power_on: True
  deploy: True
  ssh_username: root
  password: 123456
  sudo: node
  sudo_password: 123456
  minion:
    master: localhost
    make_master: True
```

Running `sudo salt-cloud -f show_image virtualbox image=Ubuntu -l debug` gives me identical output, culminating in this:

```
  File "/usr/local/lib/python2.7/dist-packages/salt/utils/virtualbox.py", line 145, in vb_get_manager
    reload(vboxapi)
UnboundLocalError: local variable 'reload' referenced before assignment
```

I'll see what I can figure out here

**Comment by waynew** [comment]
@garethgreenaway Looks like this was actually fixed in https://github.com/saltstack/salt/commit/e993692d01c9ad063db16447573e44cd40497e3f#diff-941a420c79e580f37bfbd53b3d94107a 

I'm not sure what the best approach is for the OP to solve the issue, but it looks like installing a different version (or applying that patch) should fix things.

**Comment by garethgreenaway** [comment]
@vipullinux The fix for this issue will be available in the 2018.3.4 release.  We'll close this out.  Please feel free to open additional issues if you're still having issues.  Thanks!
